src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -753,7 +753,8 @@ private BoundExpression BindDeclarationVariables(TypeSymbol declType, VariableDe
                             subExpressions.SelectAsArray(e => e.Type),
                             default(ImmutableArray<Location>),
                             default(ImmutableArray<string>),
-                            Compilation);
+                            Compilation,
+                            shouldCheckConstraints: false);
```

Comment:
FYI, this is a new tuple creation that is not checking constraints. Since this code path is failure-only and the caller is expected to produce a diagnostic., we won't check constraints further.